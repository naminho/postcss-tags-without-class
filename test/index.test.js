const fs = require('fs')
const postcss = require('postcss')
const tagsWithoutClass = require('./../index')

const applyPostCSSWithPlugin = (content, expect) => new Promise(resolve => {
  postcss([tagsWithoutClass])
    .process(content, {from: undefined})
    .then(result => resolve(result.css))
})

const getCase = (testCase) => fs.readFileSync(`./test/cases/${testCase}.css`, 'utf8')

test('adds the opt out selector to the tag', async () => {
  const result = await applyPostCSSWithPlugin('button {color: red;}')
  expect(result).toMatchSnapshot()
})

test('adds the opt out selector to the tag from external file', async () => {
    const result = await applyPostCSSWithPlugin(getCase('basic'))
    expect(result).toMatchSnapshot()
})

test('adds it to each chained tag', async () => {
  const result = await applyPostCSSWithPlugin(getCase('chained'))
  expect(result).toMatchSnapshot()
})

test('adds it to multiple tags', async () => {
  const result = await applyPostCSSWithPlugin(getCase('multiple'))
  expect(result).toMatchSnapshot()
})

test('adds it between the tag and further pseudo tags', async () => {
  const result = await applyPostCSSWithPlugin(getCase('pseudo'))
  expect(result).toMatchSnapshot()
})
