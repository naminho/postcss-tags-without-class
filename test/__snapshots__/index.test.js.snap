// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds it between the tag and further pseudo tags 1`] = `
"button:not([class]):focus {
  color: red;
}
"
`;

exports[`adds it to each chained tag 1`] = `
"a:not([class]) span:not([class]) {
  color: red;
}
"
`;

exports[`adds it to multiple tags 1`] = `
"button:not([class]),
a:not([class]) {
  color: red;
}
"
`;

exports[`adds the opt out selector to the tag 1`] = `"button:not([class]) {color: red;}"`;

exports[`adds the opt out selector to the tag from external file 1`] = `
"button:not([class]) {
    color: red;
}"
`;
